src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +45,118 @@ public void editPerson() {
         editPersonDialog.enterNewValues(newAlice);
         editPersonDialog.pressEnter();
 
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
-        //TODO: write a more elegant function that uses assertEquals but compare two different types of objects
+        //Confirm pending state correctness
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+
+        //Confirm the right card is selected after the edit
+        assertEquals(alicePersonCard, newAlice);
+        assertTrue(personListPanel.isSelected(newAlice));
 
-        //TODO: Confirm the right card is selected after the edit, as follows
-        //assertEquals(1, personListPanel.getSelectedCards().size();
-        //assertEquals(alicePersonCard, personListPanel.getSelectedCards().get(0);
 
         //Confirm right values are displayed after grace period is over
-        guiRobot.sleep(5000); //wait for grace period
-        //TODO: link up to grace period
-        //TODO: confirm the person card display behave as expected during grace period
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
+        mainGui.sleepForGracePeriod();
+        assertEquals(alicePersonCard, newAlice);
 
         //Confirm the underlying person object has the right values
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
-        //TODO: confirm again after the next sync
-        //TODO: confirm other cards are unaffected
-        //TODO: confirm the right card is selected after the edit
-        //TODO: confirm status bar is updated correctly
 
+        //Confirm again after the next sync
+        sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
+        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
+
+        //Confirm other cards are unaffected.
+        personListPanel.clickOnListView();
+        assertTrue(personListPanel.isListMatching(1, td.benson, td.charlie, td.dan, td.elizabeth));
```

Comment:
Minor thing. Create a variant of the method `isListMatching(List<Person>)` that checks for an exact match. The current one checks for sub-list match. The panel can have additional cards at the bottom and the test will still pass.